### PR TITLE
Ghost Circuits

### PR DESCRIPTION
**Pull request by Caedis**
## What
Implement ghost circuits from gt5u

## Todo

- [x] Update tooltips
- [ ] Open config gui on shift left click (deferring until MUI rework)

## Potential Compatibility Issues
Anything in the circuit slot prior to updating will be consumed and the ghost circuit will have the same config value if it had one.

### CODE REVIEW

File: src/main/java/gregtech/api/gui/widgets/GhostCircuitSlotWidget.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import net.minecraft.inventory.ClickType;
+import net.minecraft.item.ItemStack;
+import net.minecraft.network.PacketBuffer;
+import net.minecraftforge.items.IItemHandlerModifiable;
+
+
+/**
+ * Like a phantom slot, but for real fake items like circuits (and maybe molds later?)
+ */
+public class GhostCircuitSlotWidget extends SlotWidget {
```
Comment (LAGIdiot) [inline src/main/java/gregtech/api/gui/widgets/GhostCircuitSlotWidget.java:14]: All other slots with similar functionality are called `Phantom*Widget` is here some specific reason this one is named differently?

File: src/main/java/gregtech/api/metatileentity/SimpleMachineMetaTileEntity.java [hunk 2]
```diff
@@ -443,7 +449,7 @@ protected ModularUI.Builder createGuiTemplate(EntityPlayer player) {
 
         if (exportItems.getSlots() + exportFluids.getTanks() <= 9) {
             ImageWidget logo = new ImageWidget(152, 63 + yOffset, 17, 17, GTValues.XMAS.get() ? GuiTextures.GREGTECH_LOGO_XMAS : GuiTextures.GREGTECH_LOGO).setIgnoreColor(true);
-            SlotWidget circuitSlot = new SlotWidget(circuitInventory, 0, 124, 62 + yOffset, true, true, false)
+            SlotWidget circuitSlot = new GhostCircuitSlotWidget(circuitInventory, 0, 124, 62 + yOffset)
```
Comment (LAGIdiot) [inline src/main/java/gregtech/api/metatileentity/SimpleMachineMetaTileEntity.java:452]: Slot should accept only one item
image

**Review by LAGIdiot — CHANGES_REQUESTED** [review]
Implementation is mostly reasonable (some comments). In game test did not found any serious problems (small put in comment). Upgrade behavior is as described.

File: src/main/java/gregtech/api/gui/widgets/GhostCircuitSlotWidget.java (hunk 1 above)
Comment (Caedis) [reply src/main/java/gregtech/api/gui/widgets/GhostCircuitSlotWidget.java:14]: Phantom is something that can accept an item dragged in my JEI.

File: src/main/java/gregtech/api/metatileentity/SimpleMachineMetaTileEntity.java (hunk 2 above)
Comment (Caedis) [reply src/main/java/gregtech/api/metatileentity/SimpleMachineMetaTileEntity.java:452]: huh, didnt actually know the IC's stack so I didnt try it.

Comment (Caedis) [reply src/main/java/gregtech/api/metatileentity/SimpleMachineMetaTileEntity.java]: Fixed

File: src/main/java/gregtech/api/gui/widgets/GhostCircuitSlotWidget.java (hunk 1 above)
Comment (Caedis) [reply src/main/java/gregtech/api/gui/widgets/GhostCircuitSlotWidget.java]: Change it to a better name

File: src/main/java/gregtech/api/gui/widgets/GhostCircuitSlotWidget.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import gregtech.api.recipes.ingredients.IntCircuitIngredient;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.inventory.ClickType;
+import net.minecraft.item.ItemStack;
+import net.minecraft.network.PacketBuffer;
+import net.minecraftforge.items.IItemHandlerModifiable;
+
+
+/**
+ * Like a phantom slot, but for real fake items like circuits (and maybe molds later?)
```
[the remark this answers is not in the document]
Comment (Caedis) [reply src/main/java/gregtech/api/gui/widgets/GhostCircuitSlotWidget.java]: Removed

**Review by LAGIdiot — APPROVED** [review]
Thanks for finishing this PR! Approved.

This is probably the best thing we can get without getting modular UI and totally reworking GUI for this.

**Pull request merged**